src/sentry/notifications/slack.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from typing import Any, Iterable, Mapping, Optional
+
+from sentry.constants import ObjectStatus
+from sentry.integrations.slack.client import SlackClient
+from sentry.models import Integration
+from sentry.models.integration import ExternalProviders, ExternalUser, get_provider_name
+from sentry.notifications.constants import SLACK_TIMEOUT
+from sentry.shared_integrations.exceptions import ApiError
+from sentry.utils import json
+
+logger = logging.getLogger("sentry.notifications")
+
+
+def get_slack_channel_id_for_user(organization: Any, integration: Any, user: Any) -> Optional[str]:
+    """
+    For now we're assuming that the integration is ALWAYS Sentry's Slack integration.
+    We're also abusing `external_name` by expecting it to be an `external_id`.
+    TODO(mgaeta): Filter by organization, actor, and integration.
+    """
+
+    entries = ExternalUser.objects.filter(
+        organizationmember__organization=organization,
+        organizationmember__user=user,
+        provider=ExternalProviders.SLACK.value,
+    )
+    if not entries:
+        return None
+
+    return entries[0].external_name
```

Comment:
How do these external IDs map to different slack integrations?

afaict, the `ExternalUsers` table doesn't make any association to specific instances of integrations, so you could have one user with different slack external_ids, how would it know which external Id to use for which slack integration later when sending the messages?